### PR TITLE
Add compatibility with OTP 23

### PR DESCRIPTION
**Pull request by g-andrade**
Make TLS connections backed by a wildcard certificate work.
    
They otherwise wouldn't, due to an OTP [bugfix](https://bugs.erlang.org/browse/ERL-1232).

### CODE REVIEW

File: src/eredis_client.erl [hunk 1]
```diff
@@ -370,7 +370,7 @@ transport_params(State) when State#state.transport =:= tcp ->
 transport_params(State) when State#state.transport =:= ssl ->
     Host = State#state.host,
     Port = State#state.port,
-    ConnectOpts = eredis_security:secure_ssl_opts(Host, ?SSL_SOCKET_OPTS),
```
Comment (lrascao) [inline src/eredis_client.erl:373]: what other purpose does `eredis_security`serve?

**Review by lrascao — APPROVED** [review]

File: src/eredis_client.erl (hunk 1 above)
Comment (g-andrade) [reply src/eredis_client.erl:373]: None whatsoever, that's why it was removed.

Comment (lrascao) [reply src/eredis_client.erl:373]: right, missed it

File: .travis.yml [hunk 2]
```diff
@@ -3,23 +3,19 @@ sudo: required
...
-  - 17.5
-  - 17.4
-  - 17.3
-  - 17.1
-  - 17.0
-  - R16B03-1
-  - R16B03
-  - R16B02
-  - R16B01
+  - 19.3
```
Comment (paulo-ferraz-oliveira) [inline .travis.yml:7]: Why not 19.1, 19.2 and 20.2?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # CHANGELOG
 
+## [Unreleased]
+
+* Added:
+    * compatibility with OTP 23
```
Comment (paulo-ferraz-oliveira) [inline CHANGELOG.md]: No kudos?

File: src/eredis.app.src [hunk 4]
```diff
@@ -4,7 +4,6 @@
               {modules,[eredis,eredis_client,eredis_parser,eredis_sub,
                         eredis_sub_client]},
               {registered,[]},
-              {applications,[kernel,stdlib,ssl,public_key,certifi,parse_trans,
-                             ssl_verify_fun]},
+              {applications,[kernel,stdlib,ssl,tls_certificate_check]},
```
Comment (paulo-ferraz-oliveira) [inline src/eredis.app.src:7]: What happened with `public_key`? Was it not valid before?

**Review by paulo-ferraz-oliveira — APPROVED** [review]
I'm OK with the changes, even if my comments aren't resolved.

**Comment by paulo-ferraz-oliveira** [comment]
Regarding Travis CI, I did ask for it to be enabled. I just don't know how long it'll take.

File: .travis.yml (hunk 2 above)
Comment (g-andrade) [reply .travis.yml:7]: They're not available on Travis  - they stopped being available a few months ago. (Perhaps the situation has changed in the mean time.)

File: CHANGELOG.md (hunk 3 above)
Comment (g-andrade) [reply CHANGELOG.md]: I'm a selfless person, I've got no ego at all.

Fixed in ab0e350.

File: src/eredis.app.src (hunk 4 above)
Comment (g-andrade) [reply src/eredis.app.src:7]: `eredis` will no longer use it after this PR.

**Pull request merged**